src/overlays/actors/ovl_Bg_Astr_Bombwall/z_bg_astr_bombwall.c

```diff
@@ -28,59 +37,191 @@ const ActorInit Bg_Astr_Bombwall_InitVars = {
+        func_800BBFB0(globalCtx, &pos, 50.0f, 2, Rand_ZeroOne() * 120.0f + 20.0f, Rand_ZeroOne() * 240.0f + 20.0f, 0);
+        velocity.x = Rand_ZeroOne() * 2.5f;
+        velocity.y = (Rand_ZeroOne() * 2.5f) + 1.0f;
+        velocity.z = Rand_ZeroOne() * 2.5f;
+        rand = Rand_ZeroOne();
+
+        if (rand < 0.2f) {
+            var_v0 = 0x60;
+        } else if (rand < 0.6f) {
+            var_v0 = 0x40;
+        } else {
+            var_v0 = 0x20;
+        }
+        EffectSsKakera_Spawn(globalCtx, &pos, &velocity, &pos, -260, var_v0, 20, 0, 0, 10, 0, 0, 50, -1,
+                             OBJECT_ASTR_OBJ, object_astr_obj_DL_002178);
+    }
+}
+
+void func_80C0A378(BgAstrBombwall* this) {
+    this->actionFunc = func_80C0A38C;
+}
+
+void func_80C0A38C(BgAstrBombwall* this, GlobalContext* globalCtx) {
+    if (this->collider.base.acFlags & AC_HIT) {
+        this->collider.base.acFlags &= ~AC_HIT;
+        Flags_SetSwitch(globalCtx, OBJBgAstrBombwall_GET_SWITCHFLAG(&this->dyna.actor));
+        func_80C0A400(this, globalCtx);
+        return;
+    }
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+}
+
+void func_80C0A400(BgAstrBombwall* this, GlobalContext* globalCtx) {
+    this->actionFunc = func_80C0A418;
+}
+
+void func_80C0A418(BgAstrBombwall* this, GlobalContext* globalCtx) {
+    if (SubS_StartActorCutscene(&this->dyna.actor, this->cutscenes[0], -1, SUBS_CUTSCENE_SET_UNK_LINK_FIELDS)) {
+        func_80C0A458(this, globalCtx);
+    }
+}
+
+void func_80C0A458(BgAstrBombwall* this, GlobalContext* globalCtx) {
+    func_800C62BC(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    this->dyna.actor.draw = NULL;
+    func_80C0A120(this, globalCtx);
+    Actor_PlaySfxAtPos(&this->dyna.actor, NA_SE_EV_WALL_BROKEN);
+    this->actionFunc = func_80C0A4BC;
+}
+
+void func_80C0A4BC(BgAstrBombwall* this, GlobalContext* globalCtx) {
+}
+
+void BgAstrBombwall_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgAstrBombwall* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+}
+
+void BgAstrBombwall_Draw(Actor* thixs, GlobalContext* globalCtx) {
+    {
+        Gfx* opa;
+
+        OPEN_DISPS(globalCtx->state.gfxCtx);
+        opa = POLY_OPA_DISP;
+        gSPDisplayList(&opa[0], &sSetupDL[25 * 6]);
+        gSPMatrix(&opa[1], Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPSetGeometryMode(&opa[2], G_LIGHTING_POSITIONAL);
+        gSPDisplayList(&opa[3], object_astr_obj_DL_002380);
+        POLY_OPA_DISP = &opa[4];
+        CLOSE_DISPS(globalCtx->state.gfxCtx);
+    }
+    {
+        Gfx* xlu;
+
+        OPEN_DISPS(globalCtx->state.gfxCtx);
+        xlu = POLY_XLU_DISP;
+        gSPDisplayList(&xlu[0], &sSetupDL[25 * 6]);
+        gSPMatrix(&xlu[1], Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPSetGeometryMode(&xlu[2], 0x400000);
```

Comment:
```suggestion
        gSPSetGeometryMode(&xlu[2], G_LIGHTING_POSITIONAL);
```
This should also take the define.